### PR TITLE
Add "json_merge_patch" dependency

### PR DESCRIPTION
**Pull request by padraic-shafer**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -65,6 +65,7 @@ all = [
     "canonicaljson",
     "jinja2",
     "jmespath",
+    "json_merge_patch",
```
Comment (hyperrealist) [inline pyproject.toml:68]: I believe this is a duplicate as `json_merge_patch` and `json-merge-patch` refer to the same pip package. We should remove one of them and adhere to the chosen naming convention.

https://github.com/bluesky/tiled/blob/8c717a3555dc4f129fd1bd20946b9e3ced7602bf/pyproject.toml#L71

File: pyproject.toml [hunk 2]
```diff
@@ -240,6 +244,7 @@ server = [
...
```
Comment (hyperrealist) [inline pyproject.toml:247]: This one is also a duplicate of `json-merge-patch` in https://github.com/bluesky/tiled/blob/8c717a3555dc4f129fd1bd20946b9e3ced7602bf/pyproject.toml#L250

File: pyproject.toml [hunk 3]
```diff
@@ -178,6 +180,7 @@ minimal-client = [
     "appdirs",
     "entrypoints",
     "httpx >=0.20.0,!=0.23.1",
+    "json_merge_patch",
```
Comment (hyperrealist) [inline pyproject.toml:183]: Kudos for catching this one! I think we also need `jsonpatch` here.

```suggestion
    "json_merge_patch",
    "jsonpatch",
```

File: pyproject.toml [hunk 4]
```diff
@@ -201,6 +204,7 @@ minimal-server = [
     "canonicaljson",
     "jinja2",
     "jmespath",
+    "json_merge_patch",
```
Comment (hyperrealist) [inline pyproject.toml:207]: Here too, we need `jsonpatch`.

```suggestion
    "json_merge_patch",
    "jsonpatch",
```

**Review by hyperrealist — DISMISSED** [review]
Thanks for catching this @padraic-shafer! I added some review comments.

**Comment by danielballan** [comment]
I confirmed that [`json-merge-patch` with `-` not `_`](https://pypi.org/project/json-merge-patch/) is the canonical spelling on PyPI. (PyPI does redirects for `_` versus `-`.) I sorted all the deps to reduce the chance of confusion.

**Pull request merged**